projects/core/src/routing/event/routing-event.builder.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { TestBed } from '@angular/core/testing';
+import {
+  RoutingEventBuilder,
+  SearchPageVisited,
+  ProductDetailsPageVisited,
+  CategoryPageVisited,
+  HomePageVisited,
+  CartPageVisited,
+  PageVisited,
+  OrderConfirmationPageVisited,
+} from '.';
+import { ActionsSubject, Action } from '@ngrx/store';
+import { ReplaySubject, Observable, of } from 'rxjs';
+import { ProductService, ProductSearchService } from '../../product';
+import { Product } from '../../model/product.model';
+import { SemanticPathService, UrlCommands } from '../configurable-routes';
+import { EventService } from '../../event';
+import { ProductSearchPage } from '../../model';
+
+class MockProductService {
+  get(): Observable<Product> {
+    return of();
+  }
+}
+class ProductSearchServiceStub {
+  getResults(): Observable<ProductSearchPage> {
+    return of();
+  }
+}
+
+class MockSemanticPathService {
+  transform(_commands: UrlCommands): any[] {
+    return [];
+  }
+  get(_routeName: string): string {
+    return '';
+  }
+}
+
+describe('Routing-Event Builder', () => {
+  let routingEventBuilder: RoutingEventBuilder;
+  let mockActionsSubject: ReplaySubject<Action>;
+  let eventService: EventService;
+
+  const mockTearDown = () => {};
+
+  function setVariables() {
+    mockActionsSubject = new ReplaySubject<Action>();
+  }
+
+  beforeEach(() => {
+    setVariables();
+    TestBed.configureTestingModule({
+      providers: [
+        RoutingEventBuilder,
+        {
+          provide: ActionsSubject,
+          useValue: mockActionsSubject,
+        },
+        {
+          provide: EventService,
+          useValue: {
+            register: jasmine
+              .createSpy('register')
+              .and.returnValue(mockTearDown),
+          },
+        },
+        {
+          provide: ProductService,
+          useClass: MockProductService,
+        },
+        {
+          provide: ProductSearchService,
+          useClass: ProductSearchServiceStub,
+        },
+        {
+          provide: SemanticPathService,
+          useClass: MockSemanticPathService,
+        },
+      ],
+    });
+    routingEventBuilder = TestBed.inject(RoutingEventBuilder);
+    eventService = TestBed.inject(EventService);
+  });
+
+  it('should be created', () => {
+    expect(routingEventBuilder).toBeTruthy();
+  });
+
+  it('should register seven events', () => {
+    expect(eventService.register).toHaveBeenCalledTimes(7);
+  });
+
+  it('should register SearchPageVisited', () => {
+    expect(eventService.register).toHaveBeenCalledWith(
+      SearchPageVisited,
+      jasmine.any(Observable)
+    );
+  });
+
+  it('should register ProductDetailsPageVisited', () => {
+    expect(eventService.register).toHaveBeenCalledWith(
+      ProductDetailsPageVisited,
+      jasmine.any(Observable)
+    );
+  });
+
+  it('should register CategoryPageVisited', () => {
+    expect(eventService.register).toHaveBeenCalledWith(
+      CategoryPageVisited,
+      jasmine.any(Observable)
```

Comment:
Medium: Those unit tests are very minimal. It would be good to check if the event has correct data inside (i.e. for above event - the category name and code)

Reply:
It appears this has been moved to another PR